### PR TITLE
feat(mobile): refactor graph again

### PR DESCRIPTION
**Pull request by Nodonisko**
## Description

1. Refactor graph to calculate from current balance into past so we need only transactions in given timeframe
2. Fetch transaction fiat rates only on demand - probably minor/zero impact for most of user but it greatly improves app performance for users with lot of transactions
3. Bring back pagination for Transaction list
4. Added support for multiple xpubs into our Portfolio Tracker Dev Xpub
5. Use token definitions instead of fiat rates for showing tokens. This sadly introduce perf problems because coin definition lookup is very sloooow. So I replaced these selectors with ones that return static value and it affects functionality i minimal way. I will fix them in another PR because it could affect desktop too.

## Related Issue

Resolve

### CODE REVIEW

File: packages/connect-explorer/package.json [hunk 1]
```diff
@@ -1,6 +1,5 @@
 {
     "name": "@trezor/connect-explorer",
-    "packageManager": "yarn@4.0.2",
```
Comment (Nodonisko) [inline packages/connect-explorer/package.json:3]: This shoudn't be here and it breaks Nx sometimes...

Comment (mroz22) [reply packages/connect-explorer/package.json:3]: i am not sure why we have it there. Is it intended? @martykan

Comment (martykan) [reply packages/connect-explorer/package.json:3]: Probably a leftover from the Nextra template

File: suite-common/wallet-core/src/transactions/transactionsThunks.ts [hunk 2]
```diff
@@ -394,3 +397,78 @@ export const fetchAllTransactionsForAccountThunk = createSingleInstanceThunk(
         return selectAccountTransactions(getState(), accountKey);
     },
 );
+
+export const fetchTransactionsUntilTimestamp = createSingleInstanceThunk(
```
Comment (PeKne) [inline suite-common/wallet-core/src/transactions/transactionsThunks.ts]: nit: What about?
```suggestion
export const fetchTransactionsFromNowUntilTimestamp = createSingleInstanceThunk(
```

File: suite-common/wallet-core/src/transactions/transactionsThunks.ts [hunk 3]
```diff
@@ -394,3 +397,78 @@ export const fetchAllTransactionsForAccountThunk = createSingleInstanceThunk(
...
+    ) => {
+        if (!timestamp) {
+            return dispatch(fetchAllTransactionsForAccountThunk({ accountKey })).unwrap();
+        }
+        const account = selectAccountByKey(getState(), accountKey);
+        if (!account) {
+            throw new Error(`Account not found: ${accountKey}`);
+        }
+
+        const areTransactionsAlreadyFetched = selectAreAllAccountTransactionsLoadedTillTimestamp(
```
Comment (PeKne) [inline suite-common/wallet-core/src/transactions/transactionsThunks.ts]: the thunk is named `...Until...` the selector `...Till...`, Unify this please.

File: suite-common/wallet-core/src/transactions/transactionsThunks.ts [hunk 4]
```diff
@@ -394,3 +397,78 @@ export const fetchAllTransactionsForAccountThunk = createSingleInstanceThunk(
...
+
+        if (areTransactionsAlreadyFetched) {
+            return selectAccountTransactionsAfterTimestamp(getState(), accountKey, timestamp);
+        }
+
+        let page = 1;
+        let marker: AccountInfo['marker'] | undefined;
+        let totalPages = 0;
+        let forceRefetch = false;
+        const perPage = 7;
```
Comment (PeKne) [inline suite-common/wallet-core/src/transactions/transactionsThunks.ts:435]: where does this const come from? put explaining comment please

File: suite-common/wallet-core/src/transactions/transactionsThunks.ts [hunk 5]
```diff
@@ -394,3 +397,78 @@ export const fetchAllTransactionsForAccountThunk = createSingleInstanceThunk(
...
+        );
+
+        if (areTransactionsAlreadyFetched) {
+            return selectAccountTransactionsAfterTimestamp(getState(), accountKey, timestamp);
+        }
+
+        let page = 1;
+        let marker: AccountInfo['marker'] | undefined;
+        let totalPages = 0;
+        let forceRefetch = false;
```
Comment (PeKne) [inline suite-common/wallet-core/src/transactions/transactionsThunks.ts]:
```suggestion
        const forceRefetch = false;
```

File: suite-native/tokens/src/ethereumTokensSelectors.ts [hunk 6]
```diff
@@ -171,27 +162,39 @@ export const selectAccountOrTokenAccountTransactions = (
...
+        A.filter(account => isEthereumAccountSymbol(account.symbol)),
+        A.map(account => account.tokens?.map(token => token.contract)),
+        A.flat,
+        // Don't use A.uniq(By) because it's slow for large arrays
+        t => Array.from(new Set(t)),
+        A.filter(tokenAddress => !!tokenAddress),
+    ) as TokenAddress[];
+};
+
 export const selectEthereumAccountsTokensWithFiatRates = memoizeWithArgs(
```
Comment (PeKne) [inline suite-native/tokens/src/ethereumTokensSelectors.ts:179]: now it does not check the fiat rates but token definitons, right? What about `selectKnownEthereumAccountTokens`? Or something similar.

File: suite-native/tokens/src/tokensSelectors.ts [hunk 7]
```diff
@@ -1,57 +1,66 @@
...
+// };
+
+// export const selectAccountHasAnyTokensWithFiatRates = (
+//     state: AccountsRootState & TokenDefinitionsRootState,
+//     accountKey: string,
+// ) => {
+//     return selectNumberOfAccountTokensWithFiatRates(state, accountKey) > 0;
+// };
+
+// Temporary solution until we solve performance issues with token definitions
```
Comment (PeKne) [inline suite-native/tokens/src/tokensSelectors.ts]: Please move this comment at the beginning of the file, so it is obvious that the commented code should not be removed.

File: suite-native/transactions/src/hooks/useTransactionFiatRates.tsx [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+import {
+    AccountKey,
+    Timestamp,
+    TokenAddress,
+    WalletAccountTransaction,
+} from '@suite-common/wallet-types';
+import { selectFiatCurrencyCode, SettingsSliceRootState } from '@suite-native/settings';
+import { getFiatRateKey } from '@suite-common/wallet-utils';
+
+export const useTransactionFiatRates = ({
```
Comment (PeKne) [inline suite-native/transactions/src/hooks/useTransactionFiatRates.tsx]: nit:

```suggestion
export const useTransactionFiatRate = ({
```

**Review by PeKne — COMMENTED** [review]
Mostly nits. Looks promising 🤞. Did not test the performance though.

File: suite-common/wallet-core/src/transactions/transactionsThunks.ts (hunk 2 above)
Comment (Nodonisko) [reply suite-common/wallet-core/src/transactions/transactionsThunks.ts]: Yea much better

File: suite-native/tokens/src/ethereumTokensSelectors.ts (hunk 6 above)
Comment (Nodonisko) [reply suite-native/tokens/src/ethereumTokensSelectors.ts:179]: Good idea

Comment (Nodonisko) [reply suite-native/tokens/src/ethereumTokensSelectors.ts:179]: I will not rename it in this PR in the end. I am already playing with it in next PR for some selectors optimalizations.

File: suite-native/transactions/src/hooks/useTransactionFiatRates.tsx (hunk 8 above)
Comment (Nodonisko) [reply suite-native/transactions/src/hooks/useTransactionFiatRates.tsx]: fixed

File: suite-native/tokens/src/tokensSelectors.ts (hunk 7 above)
Comment (Nodonisko) [reply suite-native/tokens/src/tokensSelectors.ts]: done

File: suite-common/wallet-core/src/transactions/transactionsThunks.ts (hunk 5 above)
Comment (Nodonisko) [reply suite-common/wallet-core/src/transactions/transactionsThunks.ts]: removed it's not needed here at all

File: suite-common/wallet-core/src/transactions/transactionsThunks.ts (hunk 4 above)
Comment (Nodonisko) [reply suite-common/wallet-core/src/transactions/transactionsThunks.ts:435]: added comment

File: suite-common/wallet-core/src/transactions/transactionsThunks.ts (hunk 3 above)
Comment (Nodonisko) [reply suite-common/wallet-core/src/transactions/transactionsThunks.ts]: unified

**Review by PeKne — APPROVED** [review]

**Comment by PeKne** [comment]
LGTM. Did not tests the performance.

**Pull request merged**